File: 구본민/2기/2주차/자릿수 더하기.java
import java.util.*;

public class Solution {
    public int solution(int n) {
        //자연수 N이 주어지면, N의 각 자릿수의 합을 구해서 return 하는 solution 함수를 만들어 주세요.
        //정수형 n의 범위 : 100,000,000 이하의 자연수
        //정답이 담길 answer에 선언 및 대입
        int answer = 0;
        String numStr = String.valueOf(n);          //정수형 n을 문자열로 만들고 numStr에 대입
        String[] num = numStr.split("");            //문자열 numStr을 한글자씩 잘라서 num 문자열 배열에 대입
        for(int i=0;i<num.length;i++){              //num 배열 길이 만큼 반복문 진행
            answer += Integer.parseInt(num[i]);     //answer에 하나씩 정수형으로 돌려서 더해주기
        }
        return answer;
    }
}

File: 이준혁/2기/9주차/자릿수더하기.js
function solution(n)
{
    let answer = 0;

    while (n > 0) { // n을 10으로 계속 나눠서 마지막 자리만 더해줄 생각
        answer += n % 10; // n%10. 즉, 일의 자리만 저장해준다.
        n = (n-n%10) / 10; // n에서 일의 자리를 빼고 10으로 나눠주면 십의 자리가 일의 자리가 된다.
    }

    return answer;
}
